### PR TITLE
Clarify extension dependency instructions for adding a new language

### PR DESCRIPTION
**Pull request by fidgetingbits**
Following a discussion on Slack between Andreas and River about the Elixir PR tests failing, it seems like the docs needed to be updated to clarify how to add an extension dependency when adding support for a language type that vscode isn't natively aware of.

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [x] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: docs/contributing/adding-a-new-language.md [hunk 1]
```diff
@@ -9,7 +9,11 @@ of a document. See the
 [docs](https://github.com/pokey/vscode-parse-tree/#adding-a-new-language) there
 for how to add support for a new parser
 
-## 2. Define parse tree patterns in Cursorless
+## 2. Ensure file type is supported by VSCode
+
+If you are adding support for a new language that isn't natively detected by VSCode, you will need to add the appropriate extension to the list of dependencies. The list of languages officially supported by VSCode is listed [here](https://code.visualstudio.com/docs/languages/identifiers). You can add the extension to `packages/common/src/extensionDependencies.ts` and then re-run `pnpm init-vscode-sandbox` to ensure it is installed. If you do not do this you will ing counter errors when attempting to execute cursorless commands in the next step. See [here](https://github.com/cursorless-dev/cursorless/issues/1895)
```
Comment (r-tae) [inline docs/contributing/adding-a-new-language.md:14]: Talon typo
```suggestion
If you are adding support for a new language that isn't natively detected by VSCode, you will need to add the appropriate extension to the list of dependencies. The list of languages officially supported by VSCode is listed [here](https://code.visualstudio.com/docs/languages/identifiers). You can add the extension to `packages/common/src/extensionDependencies.ts` and then re-run `pnpm init-vscode-sandbox` to ensure it is installed. If you do not do this you will encounter errors when attempting to execute cursorless commands in the next step. See [here](https://github.com/cursorless-dev/cursorless/issues/1895)
```

**Review by r-tae — COMMENTED** [review]
LGTM apart from that typo

File: docs/contributing/adding-a-new-language.md [hunk 2]
```diff
@@ -9,7 +9,11 @@ of a document. See the
 [docs](https://github.com/pokey/vscode-parse-tree/#adding-a-new-language) there
 for how to add support for a new parser
 
-## 2. Define parse tree patterns in Cursorless
+## 2. Ensure file type is supported by VSCode
+
+If you are adding support for a new language that isn't natively detected by VSCode, you will need to add the appropriate extension to the list of dependencies. The list of languages officially supported by VSCode is listed [here](https://code.visualstudio.com/docs/languages/identifiers). You can add the extension to `packages/common/src/extensionDependencies.ts` and then re-run `pnpm init-vscode-sandbox` to ensure it is installed. If you do not do this you will encounter errors when attempting to execute cursorless commands in the next step. See [here](https://github.com/cursorless-dev/cursorless/issues/1895)
```
Comment (pokey) [inline docs/contributing/adding-a-new-language.md:14]:
```suggestion
If you are adding support for a new language that isn't natively detected by VSCode, you will need to add the appropriate extension to the list of dependencies. The list of languages officially supported by VSCode is listed [here](https://code.visualstudio.com/docs/languages/identifiers). If your language is in that list, you can skip this step and proceed to step 3. If your language is not in that list, you need to find a VSCode extension that adds support for your language, and add the id of the given extension to [`packages/common/src/extensionDependencies.ts`](../../packages/common/src/extensionDependencies.ts) and then re-run `pnpm init-vscode-sandbox` to ensure it is installed. If you do not do this you will encounter errors when attempting to execute cursorless commands in the next step. See [here](https://github.com/cursorless-dev/cursorless/issues/1895)
```

**Review by pokey — APPROVED** [review]
Made some minor tweaks but otherwise looks good

**Review by AndreasArvidsson — APPROVED** [review]

**Review by pokey — CHANGES_REQUESTED** [review]
things were kinda messed up in the deploy preview; I think I fixed it but waiting to double check

**Review by AndreasArvidsson — APPROVED** [review]

**Review by pokey — APPROVED** [review]
Ok looking good now!

**Pull request merged**